From 8fba5ad1b00b050dff50be1019c260cf180fa5b5 Mon Sep 17 00:00:00 2001
From: Vasyl Gello <vasek.gello@gmail.com>
Date: Thu, 14 Nov 2019 20:57:17 +0000
Subject: [PATCH] p7zip: Enforce large files support

The 64 bit ABI is not affected by this bug but 32-bit ABI
requires off_t64 support to be able to work with archives
containing files with total size > 2GB (2^32 bytes)

Test: in terminal app:

mkdir /sdcard/test
cd /sdcard/test
for i in `seq 1 1 21`
do
  dd if=/dev/urandom \
     of=/sdcard/test/`printf "%02d.bin" $i` \
     bs=1048576 count=100
done
7z -mx0 a test_2100.7z 01.bin 02.bin 03.bin 04.bin \
   05.bin 06.bin 07.bin 08.bin 09.bin 10.bin 11.bin \
   12.bin 13.bin 14.bin 15.bin 16.bin 17.bin 18.bin \
   19.bin 20.bin 21.bin
7z t test_2100.7z
cd /
rm -rf /sdcard/test

Change-Id: I896682163879b9f79d4072c161fba678a6677473
Signed-off-by: Vasyl Gello <vasek.gello@gmail.com>
---
 Android.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Android.mk b/Android.mk
index 15e951f..7ab6c49 100644
--- a/Android.mk
+++ b/Android.mk
@@ -11,7 +11,8 @@ common_cflags := \
     -DBREAK_HANDLER \
     -DUNICODE \
     -D_UNICODE \
-    -DUNIX_USE_WIN_FILE
+    -DUNIX_USE_WIN_FILE \
+    -D_FILE_OFFSET_BITS=64
 
 common_cflags += \
     -fexceptions \
-- 
2.11.0

